mvc_models/articles.model.js

```diff
@@ -1,5 +1,29 @@
 const db = require("../db/connection");
 
+
+exports.selectArticles = () => {
+    let articles 
+    return db.query(`
+    SELECT title, topic, author, created_at, article_id, article_img_url 
+    FROM articles
+    ORDER BY created_at DESC;`)
+    .then(({rows}) => {
+        articles = rows
+        const promisePends = []
+        rows.forEach((row) => {
+            const id = row.article_id
+            promisePends.push(db.query(`SELECT * FROM comments WHERE article_id = $1;`, [id]))
+        })
+        return Promise.all(promisePends)
+    })
+    .then((results) => {
+        for(let i = 0 ; i < results.length; i++) {
+            const element = results[i]
+            articles[i].comment_count = element.rows.length
+        }
+        return articles
+    })
+}
```

Comment:
Cleaver thinking. However, this can be achieved using only one SQL string. Think about joining the comments table on and using a count.

Reply:
Fixed nicely 👍